diff --git a/Jambase b/Jambase
index 157cd56..e7be5af 100755
--- a/Jambase
+++ b/Jambase
@@ -1002,7 +1002,7 @@ else if $(UNIX)
 
 	# UNIX defaults
 
-	CCFLAGS			?= $(CPPFLAGS) -g -DUNIX -D_THREAD_SAFE -pipe -fPIC ;	# -fwrapv ??
+	CCFLAGS			?= $(CPPFLAGS) -std=c11 -g -DUNIX -D_THREAD_SAFE -D_DEFAULT_SOURCE -pipe -fPIC ;	# -fwrapv ??
 	CCOPTFLAG		?= -O2 ;		
 	CCDEBUGFLAG		?= -g ;		
 	CCPROFFLAG		?= ;		
